### PR TITLE
Refactor-interface DB

### PR DESCRIPTION
**Pull request by ChielB**
I removed the need of the SA task by directly editing the interface table. Removed the HelloID tables from the config and create,update, delete and set permissions.

### CODE REVIEW

File: config.json
Comment (Rick-Jongbloed) [inline config.json:1]: config file looks good

Comment (Rick-Jongbloed) [inline config.json:1]: Config file looks good

File: create.ps1 [hunk 1]
```diff
@@ -59,7 +60,7 @@ try {
         #STKE_UnitOfPeriod                      = $null
     }
 
-    $queryAccountLookupSaltoSpace = "SELECT ExtID FROM [$sqlDatabaseSaltoSpace].[dbo].[tb_Users] INNER JOIN [$sqlDatabaseSaltoSpace].[dbo].[tb_Users_Ext] ON tb_Users.id_user = tb_Users_Ext.id_user WHERE $correlationAccountFieldSaltoSpace = @$correlationAccountFieldSaltoStaging"
+    #$queryAccountLookupSaltoSpace = "SELECT ExtID FROM [$sqlDatabaseSaltoSpace].[dbo].[tb_Users] INNER JOIN [$sqlDatabaseSaltoSpace].[dbo].[tb_Users_Ext] ON tb_Users.id_user = tb_Users_Ext.id_user WHERE $correlationAccountFieldSaltoSpace = @$correlationAccountFieldSaltoStaging"
```
Comment (Rick-Jongbloed) [inline create.ps1:63]: remote comment

File: create.ps1 [hunk 2]
```diff
@@ -137,8 +138,9 @@ try {
     $sqlConnection.Open()
 
     #Lookup record in Salto Space database (correlation)
+    <#
```
Comment (Rick-Jongbloed) [inline create.ps1:141]: remove comment

File: permissionSetAccessGroups.ps1 [hunk 3]
```diff
@@ -1,3 +1,4 @@
+$dryRun =$false
```
Comment (Rick-Jongbloed) [inline permissionSetAccessGroups.ps1:1]: should not be in here

File: permissionSetAccessGroups.ps1 [hunk 4]
```diff
@@ -13,21 +14,16 @@ $success = $false
...
 $queryAccountToBeProcessedBySalto = "UPDATE [$sqlDatabaseHelloId].[dbo].[$sqlDatabaseHelloIdAccountTable]  SET [ToBeProcessedBySalto] = 1 WHERE ExtUserId = @accountReference;"
 
 # Set id to the hash of the other three values to make sure a permission can only be set once
-$permissionMembership.id = -join [security.cryptography.sha256managed]::new().ComputeHash([Text.Encoding]::Utf8.GetBytes("$($permissionMembership.permissionType)_$($permissionMembership.permissionReference)_$($permissionMembership.accountReference)")).ForEach{$_.ToString("X2")}
-$queryCheckPermission = "SELECT id FROM $sqlTableMembership WHERE id=@id"
-$queryGrantPermission = "INSERT INTO $sqlTableMembership (id, permissionType, permissionReference, accountReference) VALUES (@id, @permissionType, @permissionReference, @accountReference);"
-$queryRevokePermission = "DELETE FROM $sqlTableMembership WHERE id=@id;"
+
+
+$queryCheckPermission = "SELECT [ExtZoneIDList] FROM $sqlDatabaseHelloIdAccountTable WHERE $($config.correlationAccountFieldSaltoStaging)=$aRef and  [ExtZoneIDList] like '%$($pRef.reference)%'"
```
Comment (Rick-Jongbloed) [inline permissionSetAccessGroups.ps1:24]: Lets discuss if this is the best way to do this

File: config.json
Comment (Rick-Jongbloed) [inline config.json:1]: I miss an update on the readme file

Comment (Rick-Jongbloed) [inline config.json:1]: Also don't forget to update the intake document

**Review by Rick-Jongbloed — CHANGES_REQUESTED** [review]

**Comment by rschouten97** [comment]
@ChielB  please see the comment of Rick-Jongbloed.
Please plan time for this or request this by the Github team so we can plan this.

**Comment by Rick-Jongbloed** [comment]
After checking this again, there are some changes to be made:
1. Please make sure correlation works, as the current script won't work when there are already users.
2. Please remove all references to the 'HelloID' table from each script and the configuration
3. Please make sure the table we interface with is called the staging table

**Pull request closed without merging**